server/service/user.js

```diff
@@ -0,0 +1,31 @@
+const mongoose = require('mongoose');
+const user = mongoose.model('User');
+
+exports.createUser = (data, callback) => {
+    user.create(data).then((response) => {
+        callback(null, response);
+    }, (error) => {
+        callback(error, null);
+    });
+};
+
+exports.findUser = (query, callback) => {
+    user.findOne(query, callback);
+};
+
+exports.findUsers = (query, callback) => {
+    // sorted by login property and limited users
+    user.find(query, callback).sort({ 'login': 1 }).limit(2);
```

Comment:
I imagine that you are trying to fullfill the following requirement here:

> get auto-suggest list from limit users, sorted by login property and filtered by
loginSubstring in the login property:
getAutoSuggestUsers(loginSubstring, limit)

What I understand by this requirement is that you can add two query parameter to the URL to serch users like this

`GET /api/users?login=eug&limit=20`

And it would return the first 20 users which the login contains `eug`

Reply:
Oh, I was confused by this  requirement. But your comment makes more sense :D

Reply:
I still have a problem with this part. GET /api/users?login=eug&limit=20 is a good method. But I don't now how to pass params [?login=eug&limit=20]... I did this like 

router.get('/:sort/:limit', user.findAll);

By using mongoose, I don't know how to do like [?login=eug&limit=20]

Reply:
When I do as GET /api/users/:sort/:limit
I actually get the params like
 { sort: '-1', limit: '3' }